### PR TITLE
Feat/tests

### PR DESCRIPTION
**Pull request by Pierre-Monier**
Add tests for : 

- MatomoDispatcher
- MatomoEvent
- MatomoTracker
- Session
- TraceableWidget (so inderectly TraceableWidgetMixin)
- TrackingOrderItem
- Visitor

Next step will to run test in CI and get some data about test coverage

### CODE REVIEW

File: test_ressources/mock/data.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+final mockMatomoTracker = MockMatomoTracker();
+final mockTrackingOrderItem = MockTrackingOrderItem();
+final mockHttpClient = MockHttpClient();
+final mockMatomoEvent = MockMatomoEvent();
+final mockHttpResponse = MockHttpResponse();
+final mockVisitor = MockVisitor();
+final mockSession = MockSession();
+final mockSharedPreferences = MockSharedPreferences();
+final mockPackageInfo = MockPackageInfo();
+final mockBuildContext = MockBuildContext();
```
Comment (TesteurManiak) [inline test_ressources/mock/data.dart:75]: Those might should probably be in the `mock.dart` file.

File: test_ressources/mock/data.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const matomoEventDimension = {'dimension': 'dimension'};
+const matomoDiscountAmount = 1.0;
+const matomoEventValue = 1.0;
+const matomoEventName = 'eventName';
+const matomoGoalId = 1;
+const matomoLink = 'link';
+const matomoOrderId = 'orderId';
+const matomoRevenue = 1.0;
+const matomoScreenId = '123456'; // 6 characters
+const matomoWrongScreenId = '123'; // 6 characters
```
Comment (TesteurManiak) [inline test_ressources/mock/data.dart:36]: More like 3 characters 🤣

**Review by TesteurManiak — CHANGES_REQUESTED** [review]

File: test_ressources/mock/data.dart (hunk 2 above)
Comment (Pierre-Monier) [reply test_ressources/mock/data.dart:36]: oups ^^

File: test_ressources/mock/data.dart (hunk 1 above)
Comment (Pierre-Monier) [reply test_ressources/mock/data.dart:75]: Huum why not ok

**Review by TesteurManiak — APPROVED** [review]

**Pull request merged**